### DISCUSSION TITLE
Uncaught TypeError: require2 is not a function

### DISCUSSION THREAD

**Discussion by julia-otran** [microsoft/monaco-editor#4085] [Q&A]
Hi. I'm trying to use monaco with electron and the WebpackPlugin.

However, Looking into 'ts.worker.js' compiled file there is a `require2` which is assigned as undefined:

![Screenshot from 2023-07-18 05-02-29](https://github.com/microsoft/monaco-editor/assets/3027489/13172ac6-3f1b-49ab-91e5-8c40051dbe22)

I think that's the cause of
![Screenshot from 2023-07-18 05-01-55](https://github.com/microsoft/monaco-editor/assets/3027489/58c603f5-a10b-4cc0-b2d4-37e65ea6471a)

Looking into the whole file, it only seems to have one assignment to this var, however that `void 0` does not seems correct.

I saw many people with this unexpected usage cause by a missed definition of the getWorkerUrl, however this seems to be working:

![Screenshot from 2023-07-18 05-12-46](https://github.com/microsoft/monaco-editor/assets/3027489/a20bb95b-4206-4d99-b290-864351503490)

**Comment by julia-otran** [comment]
I got to fix it.

Just including

```
    new MonacoWebpackPlugin({
      languages: ['javascript', 'typescript'],
    }),

    new NodePolyfillWebpackPlugin(),
```

**Reply by ianhattendorf** [reply]
Thank you for posting your workaround. I've narrowed this down to `new NodePolyfillWebpackPlugin({ includeAliases: ['process'] })`, my guess is Monaco was never tested with an Electron renderer with `nodeIntegration: true`.
